cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,880 @@
+        In other words, it is a graph that has no edges whose destination is not time delta 0.
+
+        Example:
+        Input graph:
+            - X1(t-2)-> X1(t-1) -> X2(t-1) -> X2(t), X1(t) -> X2(t), X1(t-1) -> X2(t-1)
+        Minimal graph:
+            - X1(t-1) -> X1(t) -> X2(t)
+        """
+        minimal_cg = TimeSeriesCausalGraph()
+
+        for edge in self.get_edges():
+            # get the relative time delta
+            assert isinstance(edge.source, TimeSeriesNode)
+            assert isinstance(edge.destination, TimeSeriesNode)
+            assert isinstance(edge.source.time_lag, int)
+            assert isinstance(edge.destination.time_lag, int)
+            assert edge.source.variable_name is not None
+            assert edge.destination.variable_name is not None
+
+            time_delta = edge.destination.time_lag - edge.source.time_lag
+            # add the edge if the time delta is 0 (no need to extract the new names)
+            if time_delta == 0 and not minimal_cg.edge_exists(
+                edge.source.variable_name, edge.destination.variable_name
+            ):
+                minimal_cg.add_edge(
+                    edge.source.variable_name,
+                    edge.destination.variable_name,
+                )
+            # otherwise if the time delta is not 0, we may have X[t-2]->X[t-1] and
+            # we must add X[t-1]->X[t]
+            else:
+                # get the new names according to the time delta
+                destination_name = get_name_with_lag(edge.destination.identifier, 0)
+                source_name = get_name_with_lag(edge.source.identifier, -time_delta)
+                if not minimal_cg.edge_exists(source_name, destination_name):
+                    # add the edge
+                    minimal_cg.add_edge(
+                        source_name,
+                        destination_name,
+                    )
+
+        return minimal_cg
+
+    def is_minimal_graph(self) -> bool:
+        """
+        Return True if the graph is minimal.
+
+        See `get_minimal_graph` for more details.
+        """
+        return self == self.get_minimal_graph()
+
+    def get_summary_graph(self) -> CausalGraph:
+        """
+        Return a summary graph.
+
+        Collapse graph in time into a single node per variable (column name).
+        This can become cyclic and bi-directed as X-1 -> Y and Y-1 -> X would become X <-> Y.
+
+        There are several cases to consider. Assume the edge in consideration is called B.
+        - if the edge is not in the summary graph, add it
+        - if there is an edge A is in the summary graph, then:
+            - if A and B have the same direction, keep the direction
+            - if A and B have different directions, make it bi-directed
+            - if one of the two is already bi-directed, keep it bi-directed
+
+        :return: The summary graph as a `CausalGraph` object.
```

Comment:
Here you have a ":return:" statement but not in "get_minimal_graph". I'd be consistent and maybe add one to the "get_minimal_graph" method too.

As an aside, please format `CausalGraph` as its appropriate hyper-reference / hyper-link